code/game/objects/items/weapons/storage/boxes.dm

```diff
@@ -623,6 +624,53 @@
 	new /obj/item/ammo_casing/shotgun/beanbag(src)
 	new /obj/item/ammo_casing/shotgun/beanbag(src)
 
+
+/obj/item/weapon/storage/box/actionfigure
+	name = "box of action figures"
+	desc = "The latest set of collectable action figures."
+	icon_state = "box"
+
+/obj/item/weapon/storage/box/actionfigure/New()
+	..()
+	for(var/i in 1 to 4)
+		var/randomFigure = pick(/obj/item/toy/figure/cmo,

```

Comment:
I copied this from the ingredients boxes below that choose random ingredients upon new().
If it works for ingredient boxes, it probably works for this?  Unless I should do it some other way.